src/routes/api/assistant/[id]/embed-snippet/+server.ts

```diff
@@ -0,0 +1,153 @@
+export async function GET({ params }) {
+	const { id } = params;
+
+	const script = `(function() {
+	function resizeIframeToContentSize(iframe) {
+		if (iframe.contentWindow) {
+			const maxHeight = window.innerHeight * 0.8; // 80% of window height
+			const chatContainerEl = iframe.contentWindow.document.getElementById('chat-container');
+			if(chatContainerEl){
+				const contentHeight = chatContainerEl.scrollHeight;
+				iframe.style.height = Math.max(400, Math.min(contentHeight, maxHeight)) + "px";
+			}
+		}
+	}
+
+	document.addEventListener('DOMContentLoaded', function() {
+		const button = document.createElement('button');
+		button.className = 'fixed z-[1002] bottom-5 right-5 z-50 px-4 gap-1 py-1 bg-black rounded-full text-white rounded cursor-pointer hover:bg-gray-800 border border-gray-200/30 transition-colors flex items-center focus:outline-none';
+
+		const img = document.createElement('img');
+		img.src = 'https://huggingface.co/chat/huggingchat/logo.svg';
+		img.alt = 'HuggingChat Logo';
+		img.className = 'size-5 mr-0.5 flex-none';
+
+		const text = document.createTextNode('Chat');
+
+		button.appendChild(img);
+		button.appendChild(text);
+
+		const modal = document.createElement('div');
+		modal.className = 'hidden fixed inset-0 z-[1001] overflow-auto bg-black bg-opacity-50';
+
+		const modalContent = document.createElement('div');
+		modalContent.className = 'bg-white  max-w-2xl rounded-xl overflow-hidden bottom-16 right-5 absolute max-sm:left-5 sm:w-[460px] shadow-2xl';
+
+		const iframe = document.createElement('iframe');
+		iframe.className = 'w-full';
+		iframe.style.height = '400px'; // Set an initial height
+		iframe.src = \`http://localhost:5173/chat/?embeddedAssistantId=${id}\`;
```

Comment:
this will have to be configurable using the base path I guess in order to work on self-hosted versions